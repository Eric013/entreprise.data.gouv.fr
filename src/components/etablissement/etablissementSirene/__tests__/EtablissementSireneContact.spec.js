import EtablissementSireneContact from '@/components/etablissement/etablissementSirene/EtablissementSireneContact.vue'
import { createLocalVue, shallow, mount } from '@vue/test-utils'
import { __createMocks as createStoreMocks } from '@/store/index.js'
import Vuex from 'vuex'

const localVue = createLocalVue()
localVue.use(Vuex)
jest.mock('@/store/index.js')

describe('EtablissementSireneContact.vue', () => {
  let storeMocks
  let wrapperEPC

  beforeEach(() => {
    storeMocks = createStoreMocks()
  
    wrapperEPC= shallow(EtablissementSireneContact, {
      localVue,
      store: storeMocks.store
    })
  })

  test('Computed value formattedDate returns null if no date is present', () => {
    wrapperEPC.vm.date_creation = null
    wrapperEPC.update()
    expect(wrapperEPC.vm.formattedDate).toBe(null)
  })

  test('Computed value formattedDate compute the correct date if present', () => {
    wrapperEPC.vm.result.date_creation = '20180201'
    wrapperEPC.update()
    expect(wrapperEPC.vm.formattedDate).toBe('01/02/2018')
  })

  test('it should compute the full name correctly if name and surname are present', () => {
    wrapperEPC.vm.result.nom = 'Forever'
    wrapperEPC.vm.result.prenom = 'Michel'
    wrapperEPC.update()
    expect(wrapperEPC.vm.fullOwnerName).toEqual('Forever Michel')
  })

  test('it should display the name if only the name is present', () => {
    wrapperEPC.vm.result.nom = 'Forever'
    wrapperEPC.vm.result.prenom = null
    wrapperEPC.update()
    expect(wrapperEPC.vm.fullOwnerName).toEqual('Forever')
  })

  test('it shouldnt display the name if only the first name is present', () => {
    wrapperEPC.vm.result.nom = null
    wrapperEPC.vm.result.prenom = 'Michel'
    wrapperEPC.update()
    expect(wrapperEPC.vm.fullOwnerName).toEqual('')
  })
})

describe('EtablissementSireneContact.vue snapshot', () => {

  const localVue = createLocalVue()
  const storeMocks = createStoreMocks()
  const wrapperEPC = mount(EtablissementSireneContact, {
    localVue,
    store: storeMocks.store
  })

  test('It match the snapshot', () => {
    expect(wrapperEPC.vm.$el).toMatchSnapshot()
  })
})