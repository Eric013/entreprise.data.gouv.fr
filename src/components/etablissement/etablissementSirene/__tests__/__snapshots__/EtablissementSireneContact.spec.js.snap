// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EtablissementSireneContact.vue snapshot It match the snapshot 1`] = `
<div
  class="company__panel panel"
>
  <div
    class="company__item"
  >
    <div
      class="company__item-key"
    >
      Gérant
    </div>
    <div
      class="company__item-value"
    >
       Non renseigné
    </div>
  </div>
   
  <div
    class="company__item"
  >
    <div
      class="company__item-key"
    >
      Adresse
    </div>
    <div
      class="company__item-value"
    >
        
    </div>
  </div>
   
  <div
    class="company__item"
  >
    <div
      class="company__item-key"
    >
      Ville
    </div>
    <div
      class="company__item-value"
    >
        
    </div>
  </div>
   
  <div
    class="company__item"
  >
    <div
      class="company__item-key"
    >
      Cedex
    </div>
    <div
      class="company__item-value"
    >
       Non renseigné
    </div>
  </div>
   
  <div
    class="company__item"
  >
    <div
      class="company__item-key"
    >
      Date de création
    </div>
    <div
      class="company__item-value"
    >
       
    </div>
  </div>
   
  <div
    class="company__item"
  >
    <div
      class="company__item-key"
    >
      Téléphone
    </div>
    <div
      class="company__item-value"
    >
       Non renseigné
    </div>
  </div>
   
  <div
    class="company__item"
  >
    <div
      class="company__item-key"
    >
      Email
    </div>
    <div
      class="company__item-value"
    >
       Non renseigné
    </div>
  </div>
   
  <div
    class="company__item"
  >
    <div
      class="company__item-key"
    >
      Tranche d'effectif salariés
    </div>
    <div
      class="company__item-value"
    >
       
    </div>
  </div>
</div>
`;
